### PR TITLE
[ISSUE #5026] add timer message put result process

### PR DESCRIPTION
**Pull request by yiduwangkai**
add accurate delay message return results process

### CODE REVIEW

**Review by RongtongJin — APPROVED** [review]

**Review by Oliverwqcwrw — APPROVED** [review]

File: broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java [hunk 1]
```diff
@@ -351,6 +351,21 @@ private RemotingCommand handlePutMessageResult(PutMessageResult putMessageResult
                 response.setRemark(String.format("the message is illegal, maybe msg body or properties length not matched. msg body length limit %dB, msg properties length limit 32KB.",
                     this.brokerController.getMessageStoreConfig().getMaxMessageSize()));
                 break;
+            case WHEEL_TIMER_MSG_ILLEGAL:
+                response.setCode(ResponseCode.WHEEL_TIMER_MSG_ILLEGAL);
+                response.setRemark(String.format("the accurate delay message is illegal, delay time max limit %dms, or check props if setting TIMER_DEL_UNIQKEY or not",
+                        this.brokerController.getMessageStoreConfig().getTimerMaxDelaySec() * 1000));
+                break;
```
Comment (RongtongJin) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java:358]: How about reusing ResponseCode.MESSAGE_ILLEGAL for WHEEL_TIMER_MSG_ILLEGAL？

File: broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java [hunk 2]
```diff
@@ -351,6 +351,21 @@ private RemotingCommand handlePutMessageResult(PutMessageResult putMessageResult
...
+            case WHEEL_TIMER_FLOW_CONTROL:
+                response.setCode(ResponseCode.WHEEL_TIMER_FLOW_CONTROL);
+                response.setRemark(String.format("or send message too much, triggered the flow control max num limit is %d or the current value is greater than %d and less than %d, trigger random flow control",
+                        this.brokerController.getMessageStoreConfig().getTimerCongestNumEachSlot() * 2L, this.brokerController.getMessageStoreConfig().getTimerCongestNumEachSlot(), this.brokerController.getMessageStoreConfig().getTimerCongestNumEachSlot() * 2L));
+                break;
+            case WHEEL_TIMER_NOT_ENABLE:
+                response.setCode(ResponseCode.WHEEL_TIMER_NOT_ENABLE);
+                response.setRemark(String.format("accurate delay is not enabled, timerWheelEnable is %s",
+                        brokerController.getMessageStoreConfig().isTimerWheelEnable()));
```
Comment (RongtongJin) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java:367]: How about reusing ResponseCode.SYSTEM_ERROR for WHEEL_TIMER_FLOW_CONTROL and WHEEL_TIMER_NOT_ENABLE?

File: broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java [hunk 3]
```diff
@@ -351,6 +351,21 @@ private RemotingCommand handlePutMessageResult(PutMessageResult putMessageResult
                 response.setRemark(String.format("the message is illegal, maybe msg body or properties length not matched. msg body length limit %dB, msg properties length limit 32KB.",
                     this.brokerController.getMessageStoreConfig().getMaxMessageSize()));
                 break;
+            case WHEEL_TIMER_MSG_ILLEGAL:
+                response.setCode(ResponseCode.WHEEL_TIMER_MSG_ILLEGAL);
+                response.setRemark(String.format("the accurate delay message is illegal, delay time max limit %dms, or check props if setting TIMER_DEL_UNIQKEY or not",
```
Comment (RongtongJin) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java:356]: Maybe "timer message illegal, the delay time should not be bigger than the max delay; or if set del msg, the delay time should be bigger than the current time" better

File: common/src/main/java/org/apache/rocketmq/common/protocol/ResponseCode.java [hunk 4]
```diff
@@ -92,6 +92,12 @@ public class ResponseCode extends RemotingSysResponseCode {
 
     public static final int FLOW_CONTROL = 215;
 
+    public static final int WHEEL_TIMER_MSG_ILLEGAL = 216;
+
+    public static final int WHEEL_TIMER_FLOW_CONTROL = 217;
+
+    public static final int WHEEL_TIMER_NOT_ENABLE = 218;
+
```
Comment (RongtongJin) [inline common/src/main/java/org/apache/rocketmq/common/protocol/ResponseCode.java:100]: IMO, It would be better not to add a new response code

**Review by RongtongJin — CHANGES_REQUESTED** [review]

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**